src/components/InfiniteScroll/index.tsx

```diff
@@ -0,0 +1,211 @@
+    if (!infiniteScrollState.loadingMore && !infiniteScrollState.allDataFetched) {
+      setInfiniteScrollState({
+        ...infiniteScrollState,
+        loadingMore: true,
+      });
+
+      const lastDataElement = infiniteScrollState.data[infiniteScrollState.data.length - 1];
+      const { page } = infiniteScrollState;
+
+      dataFetchQuery(lastDataElement, page, contentBatchSize)
+        .then((response) => {
+          const contentReceived = response;
+
+          if (contentReceived.length === 0) {
+            setInfiniteScrollState({
+              ...infiniteScrollState,
+              allDataFetched: true,
+              loadingMore: false,
+              error: null,
+            });
+          } else {
+            setInfiniteScrollState({
+              ...infiniteScrollState,
+              data: infiniteScrollState.data.concat(contentReceived),
+              error: null,
+              loadingMore: false,
+              page: infiniteScrollState.page + 1,
+            });
+          }
+        })
+        .catch((err) => {
+          setInfiniteScrollState({
+            ...infiniteScrollState,
+            allDataFetched: true,
+            error: err,
+            loadingMore: false,
+          });
+        });
+    }
+  }
+
+  function renderError() : JSX.Element | null {
+    if (infiniteScrollState.error != null) {
+      return (
+        <TextContainer style={{ ...errorContainerStyles }}>
+          <ErrorMessage>
+            Erro ao buscar dados: {infiniteScrollState.error}
+          </ErrorMessage>
+        </TextContainer>
+      );
+    }
+
+    return null;
+  }
+
+  function renderLoading() : JSX.Element {
+    if (infiniteScrollState.allDataFetched) {
+      return (
+        <TextContainer>
+          <NoDataFoundMessage>Não há dados para exibir!</NoDataFoundMessage>
+        </TextContainer>
+      );
+    }
+
+    return (
+      <LoadingContainer style={{ ...loadingContainerStyles }}>
+        <ActivityIndicator size="large" color={activityIndicatorColor} />
+      </LoadingContainer>
+    );
+  }
+
+  function renderLoadingMore() : JSX.Element | null {
+    if (infiniteScrollState.loadingMore) return renderLoading();
+
+    return null;
+  }
+
+  // Component effects.
+  useEffect(componentDidMount, [fetchInitialData]);
+
+  // JSX returned.
+  return (
+    <FlatList
+      numColumns={numColumns}
+      contentContainerStyle={{ ...contentContainerStyles }}
+      data={infiniteScrollState.data}
+      initialNumToRender={contentBatchSize}
+      keyExtractor={(item) => keyExtractorFunction(item)}
+      ListEmptyComponent={renderLoading}
+      ListFooterComponent={renderLoadingMore}
+      ListHeaderComponent={renderError}
+      nestedScrollEnabled
+      onEndReached={fetchMoreData}
+      onEndReachedThreshold={0.1}
+      renderItem={({ item }) => formatContent(item)}
+    />
+  );
+};
+
+// Default props.
+InfiniteScroll.defaultProps = { ...defaultProps, numColumns: 1 };
```

Comment:
Ok, this is pure nitpicking on my part, but, if we have a defaultProps object, shouldn't we put the numColumns default there?

Reply:
Solved.